app/utils/chart-config/versioning.ts

```diff
@@ -0,0 +1,148 @@
+export const CHART_CONFIG_VERSION = "1.0.2";
+
+const migrations = [
+  // MAP
+  // baseLayer {
+  //   + locked
+  //   + bbox
+  // }
+  {
+    from: "1.0.0",
+    to: "1.0.1",
+    up: (config: any) => {
+      let newConfig = { ...config, version: "1.0.1" };
+
+      if (newConfig.chartType === "map") {
+        const { baseLayer } = newConfig;
+        newConfig = {
+          ...newConfig,
+          baseLayer: {
+            show: baseLayer.show,
+            locked: false,
+          },
+        };
+      }
+
+      return newConfig;
+    },
+    down: (config: any) => {
+      let newConfig = { ...config, version: "1.0.0" };
+
+      if (newConfig.chartType === "map") {
+        const { baseLayer } = newConfig;
+        newConfig = {
+          ...newConfig,
+          baseLayer: {
+            show: baseLayer.show,
+          },
+        };
+      }
+
+      return newConfig;
+    },
+  },
+  // MAP
+  // symbolLayer {
+  //   + colors {
+  //     + type
+  //     + value
+  //     + opacity
+  //   }
+  //   - color
+  // }
+  {
+    from: "1.0.1",
+    to: "1.0.2",
+    up: (config: any) => {
+      let newConfig = { ...config, version: "1.0.2" };
+
+      if (newConfig.chartType === "map") {
+        const { fields } = newConfig;
+        const { symbolLayer } = fields;
+        newConfig = {
+          ...newConfig,
+          fields: {
+            ...fields,
+            symbolLayer: {
+              show: symbolLayer.show,
+              componentIri: symbolLayer.componentIri,
+              measureIri: symbolLayer.measureIri,
```

Comment:
Nit: Spreading the symbolLayer could even be clearer ? Actually I think that using immer could be perfect.